pkg/client/group_mgmt_client.go

```diff
@@ -358,42 +296,66 @@ func (client *GroupMgmtClient) listGet(
 	if err != nil {
 		return nil, err
 	}
+	return processResponse(client, response, path, nil, params)
+}
 
-	// http Error
-	if !response.IsSuccess() {
-		errResp, err := unwrapError(response.Body())
-		if err != nil {
-			return nil, err
-		}
-		return nil, fmt.Errorf("http response error: status (%d), messages: %v", response.StatusCode(), errResp)
-	}
-
+// unwrapData
+func unwrapData(body []byte, payload interface{}) (interface{}, *int, error) {
 	// unmarshal the response
-	wrapper := &DataWrapper{}
-	err = json.Unmarshal(response.Body(), wrapper)
+	wrapper := &DataWrapper{
+		Data: payload,
+	}
+	err := json.Unmarshal(body, wrapper)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
-
 	// return it
-	return wrapper, nil
+	return wrapper.Data, wrapper.TotalRows, nil
 }
 
-// unwrap a response body
-func unwrap(body []byte, payload interface{}) (interface{}, error) {
-	// TODO: add some logging
+//processResponse
+func processResponse(client *GroupMgmtClient, response *resty.Response, path string, respHolder interface{}, params *param.GetParams) (interface{}, error) {
 
-	// unmarshal the response
-	wrapper := &DataWrapper{
-		Data: payload,
+	// process successfull response
+	if response.IsSuccess() {
+		// http code 202, handle async job
+		if response.StatusCode() == 202 {
+			id, err := processAsyncResponse(client, response.Body())
+			if id != nil {
+				// action rpc may contain different path.
+				// extract Get uri path from original path.
+				newPath := strings.Split(path, "/")
+				return client.Get(newPath[0], id.(string), respHolder)
+			} else {
+				return nil, err
+			}
+		}
+		// unmarshall response data
```

Comment:
The status code is presumed to be 200 OK when we get here, right?  I suggest checking for that and returning some kind of "unexpected status code" error otherwise.

Reply:
Not require since error is already captured in err

Reply:
I'm referring to the handling of status codes like 201 Created, 204 No Content, etc.